### DISCUSSION TITLE
mcuboot upgrade path

### DISCUSSION THREAD

**Discussion by macklenc** [mcu-tools/mcuboot#945] [Q&A] [answered]
Hi!

I'm looking into swapping my project over to using mcuboot for the bluetooth firmware upgrade options it provides, but I can't seem to find any information on whether or not the bootloader itself can be upgraded over-the-air. Does mcuboot support upgrading itself?

**Comment by nvlsianpu** [comment, marked as answer]
MCUboot itself doesn't support self upgrade. However this is possible in other projects which uses MCUboot as the second stage bootloader: for sure nRFConnect SDK multistage bootloader solution is able to update MCUboot, Probably TFM project have also this ability (but I'm not sure).

**Reply by macklenc** [reply]
Gotcha, I guess that explains why support for upgrading the bootloader is dictated elsewhere (nRF 52 in this case apparently doesn't support it): https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/ug_nrf52.html#fota-upgrades.

Thanks!

**Reply by nvlsianpu** [reply]
nRF52 can run Immutable bootloader https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/samples/bootloader/README.html which does support B1 update. B1 is slightly modified MCUboot.
see also https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/ug_bootloader.html#upgradable-bootloader

**Reply by macklenc** [reply]
Awesome, thanks for the fast replies!
